src/IO/ReadBufferFromS3.cpp

```diff
@@ -197,7 +197,7 @@ size_t ReadBufferFromS3::readBigAt(char * to, size_t n, size_t range_begin, cons
             ProfileEvents::increment(ProfileEvents::ReadBufferFromS3Bytes, bytes_copied);
 
             if (cancelled)
-                return initial_n - n;
+                return initial_n - n - bytes_copied;
```

Comment:
I think it should be `initial_n - (n - bytes_copied)` because we return the number of bytes that were actually read

Reply:
ah, yeah you are right, math can be hard sometimes
thanks for catching it!